File: Data Structures & Algorithms/permutations/submission-2.py
class Solution:
    def permute(self, nums: List[int]) -> List[List[int]]:
        
        res = []
        visit = [0]*len(nums)

        def dfs(path,visit):
            if len(path) == len(nums):
                res.append(path.copy())
                return
            for i in range(len(nums)):
                if visit[i]==1:
                    continue
                path.append(nums[i])
                visit[i]=1
                dfs(path,visit)
                path.pop()
                visit[i]=0

        dfs([], visit)
        return res

File: Data Structures & Algorithms/three-integer-sum/submission-2.cpp
class Solution {
public:
    vector<vector<int>> threeSum(vector<int>& nums) {
        int n = nums.size();
        int j =0,k=0;
        sort(nums.begin(),nums.end());
        set<vector<int>> res;
        for (int i=0;i<n;i++){
            for(int j =i+1;j<n;j++){
                for(int k = j+1;k<n;k++){
                    if(nums[i]+nums[j]+nums[k]==0){
                        res.insert({nums[i],nums[j],nums[k]});
                    }
                }
            }
            // j = i+1;
            // k = j+1;
            // while(j<n-1){
            //     k = j+1;
            //     while(k<n){
            //         if(nums[i]+nums[j]+nums[k]==0){
            //             cout << nums[i] << nums[j] << nums[k]<< endl;
            //             res.push_back({nums[i],nums[j], nums[k]});
            //         }
            //         k++;
            //     }
            //     j++;
            // }
            // return res;
        }
        return vector<vector<int>>(res.begin(),res.end());
    }
};
